### PR TITLE
Add PartialEq and Eq derives

### PR DESCRIPTION
**Pull request by genusistimelord**
Add PartialEq and Eq setups to RangeSet, RangeMap, RangeInclusiveSet and RangeInclusiveMap. This is being added so we can use this library with cosmic_text for AttrList range handling.  If you could get this in a published crate that would help us out tremendously.  This is for #https://github.com/pop-os/cosmic-text/issues/28

### CODE REVIEW

**Comment by genusistimelord** [comment]
@jeffparsons  could you review this?

**Comment by jeffparsons** [comment]
I'm afraid I'm unwell right now. I will get to this ASAP when better.

**Comment by jeffparsons** [comment]
Quick general comment while I'm here: I'd prefer to avoid using a `derive` for this, mostly because adding the `K: Eq` bound to the struct itself is technically a breaking change, even if it's unlikely to affect many realistic uses of the crate.

**Comment by jeffparsons** [comment]
Come to think of it, it may well affect totally normal use cases. E.g. if someone has built a type that uses `RangeMap` internally, and they've also preferred to put trait bounds on impls where possible rather than the struct itself (I believe this pattern is established by the standard library, so I'd expect it to be common),

**Comment by genusistimelord** [comment]
I have added in a manually PartialEq, Eq and even added Ord and PartialOrd while I was at it.  @jeffparsons thank you for getting back with me even though you are ill. I hope you feel better soon.

File: src/inclusive_map.rs [hunk 1]
```diff
@@ -48,6 +49,45 @@ where
     }
 }
 
+impl<K, V> PartialEq for RangeInclusiveMap<K, V, K>
```
Comment (jeffparsons) [inline src/inclusive_map.rs:52]: These impls should have parameters like this:

```
impl<K, V, StepFnsT> RangeInclusiveMap<K, V, StepFnsT>
where
    K: Ord + Clone,
    V: Eq + Clone,
    StepFnsT: StepFns<K>,
{
```

I.e. StepFnsT shouldn't be hard-coded to be the same as K.

**Comment by jeffparsons** [comment]
Otherwise looks great to me!

Thanks very much for your PR, @genusistimelord !

**Comment by genusistimelord** [comment]
ok I think I did that correct please give it one more check @jeffparsons

**Comment by jeffparsons** [comment]
Thanks @genusistimelord . I expect to merge this and cut a release this weekend.

**Comment by genusistimelord** [comment]
nice! thank you very much.

**Comment by jeffparsons** [comment]
@genusistimelord Release containing your changes is now on crates.io

Enjoy responsibly! :stuck_out_tongue:

**Pull request merged**